### PR TITLE
Do not raise a 415 error when no content-type and no body

### PR DESCRIPTION
**Pull request by leplatrem**
Return a 415 error response, if and only if there is some content.

@almet r?

### CODE REVIEW

**Comment by Natim** [comment]
Refs https://github.com/mozilla-services/cornice/issues/353

**Comment by Natim** [comment]
I did some manual checking with Kinto and the behavior is the one we want :+1:

File: cornice/pyramidhook.py [hunk 1]
```diff
@@ -219,7 +219,8 @@ def register_service_views(config, service):
         predicate_definitions = _pop_complex_predicates(args)
 
         if predicate_definitions:
-            for predicate_list in predicate_definitions:
+            extra_empty_ctype = [({'kind': 'content_type', 'value': ''},)]
```
Comment (almet) [inline cornice/pyramidhook.py]: Ctype [refers to something different](https://docs.python.org/2/library/ctypes.html) in the python context.

**Comment by almet** [comment]
That's indeed a better behaviour. We should change the naming, but apart from that it looks good to land!

**Pull request merged**